shared/process-tc-artifacts.sh

```diff
@@ -47,8 +47,10 @@ tar -x -z -C generated-$PLATFORM -f $PLATFORM.generated-files.tar.gz
 
 date
 
+NORMALIZE_EXPR="s#/builds/worker/checkouts/gecko/##g;s#z:/task_[0-9]*/build/src/##gI"
+
 # Run the rust analysis here.
-$MOZSEARCH_PATH/scripts/rust-analyze.sh $CONFIG_FILE $TREE_NAME $PLATFORM
+$MOZSEARCH_PATH/scripts/rust-analyze.sh $CONFIG_FILE $TREE_NAME $PLATFORM "$NORMALIZE_EXPR"
```

Comment:
I'd actually prefer to do the normalization right here if possible, instead of having `rust-analyze.sh` do it. Would that be possible, or are there complications with that?

Reply:
My rationale was that:
- process-tc-artifacts.sh is happening in parallel across all platforms, including the unzipping of the rust artifacts into the objdir
- As part of the changes I made in https://bugzilla.mozilla.org/show_bug.cgi?id=1583031 there's now a bunch of path-logic in https://github.com/mozsearch/mozsearch/blob/master/scripts/rust-analyze.sh and I didn't want to end up duplicating that again in process-tc-artifacts.sh. (And more naive options would perhaps result in the parallel per-platform jobs stomping on each other.)

Perhaps the right thing for us to do is to simplify things by unzipping the rust files into per-platform directories.  The normalization step can then be run naively across the entire directory tree, as can the rust-analyze.sh step.  I think I was perhaps too overly conservative in trying to perturb the existing pipeline when attempting to address bug 1583031.